pandas/tests/copy_view/test_constructors.py

```diff
@@ -179,3 +179,20 @@ def test_dataframe_from_dict_of_series_with_dtype(index):
     df.iloc[0, 0] = 100
     arr_after = get_array(df, "a")
     assert np.shares_memory(arr_before, arr_after)
+
+
+@pytest.mark.parametrize("copy", [False, None, True])
+def test_frame_from_numpy_array(using_copy_on_write, copy, using_array_manager):
+    arr = np.array([[1, 2], [3, 4]])
+    df = DataFrame(arr, copy=copy)
+
+    if (
+        using_copy_on_write
+        and copy is not False
+        or copy is True
+        or using_array_manager
+        and copy is None
```

Comment:
```suggestion
        or (using_array_manager and copy is None)
```

I don't know if black is fine with it this way, but it would make it a bit more readable (understand you have to mentally add brackets anyway to see which and/or parts belong together)